src/lxc/mainloop.c

```diff
@@ -84,6 +84,8 @@ int lxc_mainloop_add_handler(struct lxc_epoll_descr *descr, int fd,
 	struct epoll_event ev;
 	struct mainloop_handler *handler;
 	struct lxc_list *item;
+	if (fd < 0)
+		return 0;
```

Comment:
This should probably log at least a warning.

Why do you want to return success and not an error?

Reply:
accept your suggestion for return -1, but don't add error print,because when every call lxc_mainloop_add_handler function, it had checked the return value